plugins/permission-react/api-report.md

```diff
@@ -44,17 +45,31 @@ export const permissionApiRef: ApiRef<PermissionApi>;
 // @public
 export const PermissionedRoute: (
   props: ComponentProps<typeof Route> & {
-    permission: Permission;
-    resourceRef?: string;
     errorComponent?: ReactElement | null;
-  },
+  } & (
+      | {
+          permission: Exclude<Permission, ResourcePermission>;
+          resourceRef?: never;
+        }
+      | {
+          permission: ResourcePermission;
+          resourceRef: string | undefined;
+        }
+    ),
 ) => JSX.Element;
 
 // @public
-export const usePermission: (
-  permission: Permission,
-  resourceRef?: string | undefined,
-) => AsyncPermissionResult;
+export function usePermission(
+  input:
+    | {
+        permission: Exclude<Permission, ResourcePermission>;
+        resourceRef?: never;
+      }
+    | {
+        permission: ResourcePermission;
+        resourceRef: string | undefined;
+      },
+): AsyncPermissionResult;
```

Comment:
Did you consider and nope out of the `usePermission(permission, {resourceRef})` route? Feeling that `usePermission({permission})` is a tiny bit awkward, but really no biggie

Reply:
A prior version of this code used a tuple type to allow `usePermission(permission: ResourcePermission, resourceRef?: string)` and `usePermission(permission: Exclude<Permission, ResourcePermission>)`, and we switched over to an object in [3a9db9d](https://github.com/backstage/backstage/pull/10067/commits/3a9db9d1de3236e7acf9a9602007e5fe879138ed) to make the types a bit simpler. I did play around with accepting an unwrapped non-resource permission or an object, but it didn't come out cleanly - if we want to get back to `usePermission(permission)` I'd personally be most tempted to revert the above commit to go back to the tuple.

Reply:
Sounds like you did exactly that :grin: (investigate and nope out), let's keep as is 👍